tests/test_resolvers.py

```diff
@@ -0,0 +1,237 @@
+"""Tests for multiaddr resolvers."""
+
+import sys
+from unittest.mock import AsyncMock, patch
+
+import dns.resolver
+import pytest
+import trio
+
+from multiaddr import Multiaddr
+from multiaddr.exceptions import RecursionLimitError
+from multiaddr.resolvers import DNSResolver
+
+if sys.version_info >= (3, 11):
+    from builtins import BaseExceptionGroup
+else:
+    class BaseExceptionGroup(Exception):
+        pass
+
+
+@pytest.fixture
+def dns_resolver():
+    """Create a DNS resolver instance."""
+    return DNSResolver()
+
+
+@pytest.mark.trio
+async def test_resolve_non_dns_addr(dns_resolver):
+    """Test resolving a non-DNS multiaddr."""
+    ma = Multiaddr("/ip4/127.0.0.1/tcp/1234")
+    result = await dns_resolver.resolve(ma)
+    assert result == [ma]
+
+
+@pytest.mark.trio
+async def test_resolve_dns_addr(dns_resolver):
+    """Test resolving a DNS multiaddr."""
+    # Create mock DNS answer for A record (IPv4)
+    mock_answer_a = AsyncMock()
+    mock_rdata_a = AsyncMock()
+    mock_rdata_a.address = "127.0.0.1"
+    mock_answer_a.__iter__.return_value = [mock_rdata_a]
+
+    # Create mock DNS answer for AAAA record (IPv6) - return empty to avoid conflicts
+    mock_answer_aaaa = AsyncMock()
+    mock_answer_aaaa.__iter__.return_value = []
+
+    with patch.object(dns_resolver._resolver, 'resolve') as mock_resolve:
+        # Configure the mock to return different results based on record type
+        async def mock_resolve_side_effect(hostname, record_type):
+            if record_type == "A":
+                return mock_answer_a
+            elif record_type == "AAAA":
+                return mock_answer_aaaa
+            else:
+                raise dns.resolver.NXDOMAIN()
```

Comment:
The mock code is repeated a lot. Probably worth pulling it out to a fixture.

Reply:
Sure, refactored the duplicate mock DNS resolution code into a reusable fixture